shell/src/shell.cpp

```diff
@@ -0,0 +1,36 @@
+#include "shell/src/shell.h"
+
+namespace shell {
+
+void Shell::start() {
+
+    status_ = ShellStatus::RUNNING;
+    State state;
+    while (is_running()) {
+        std::string cmd_line;
+        std::getline(streams_->get_in_stream(), cmd_line);
+        if (cmd_line.empty()) {
+            this->stop();
+            return;
+        }
+
+        parser::Parser cmd_parser{cmd_line};
+
+        StreamsBuffered streams_local;
+
+        while (true) {
+            auto descriptor = cmd_parser.parse_command();
+            if (descriptor == std::nullopt) {
+                break;
+            }
+
+            /* TODO: command processor should be there*/
+
+            descriptor.value().command->execute(descriptor.value().arguments, state, streams_local);
```

Comment:
looks like a little bit complex, and also Demetre's law 
Suggest to split into logical part and process them separately:

command from parser
substitution vars // think how to
reparsing 
execution

Reply:
Demetre's law nice explanation https://www.youtube.com/watch?v=JTLO39KwPWQ